akka-persistence/src/main/scala/akka/persistence/journal/JournalProxy.scala

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.persistence.journal
+
+import akka.util.Helpers.Requiring
+import scala.concurrent.duration._
+import akka.actor.Actor
+import akka.actor.Stash
+import scala.concurrent.duration.FiniteDuration
+import akka.actor.ActorRef
+import akka.persistence.JournalProtocol
+import akka.actor.ActorSystem
+import akka.persistence.Persistence
+import scala.util.control.NoStackTrace
+import java.util.concurrent.TimeoutException
+import akka.persistence.AtomicWrite
+import akka.persistence.NonPersistentRepr
+import akka.persistence.DeleteMessagesFailure
+import akka.actor.ActorLogging
+import com.typesafe.config.Config
+import akka.actor.Address
+import akka.actor.ActorIdentity
+import akka.actor.RootActorPath
+import akka.actor.Identify
+import akka.actor.ReceiveTimeout
+import akka.actor.ExtendedActorSystem
+import akka.persistence.SaveSnapshotFailure
+import akka.persistence.DeleteSnapshotFailure
+import akka.persistence.DeleteSnapshotsFailure
+import akka.persistence.SnapshotProtocol
+
+object JournalProxy {
+  final case class TargetLocation(address: Address)
+  private case object InitTimeout
+
+  def setTargetLocation(system: ActorSystem, address: Address): Unit = {
+    Persistence(system).journalFor(null) ! TargetLocation(address)
+    if (system.settings.config.getString("akka.persistence.snapshot-store.plugin") != "")
+      Persistence(system).snapshotStoreFor(null) ! TargetLocation(address)
```

Comment:
so this sets both, snapshot store and journal to the same address? I guess this is indeed the most popular use case for this utility.
In theory could be split up, but let's keep as is - document what it sets on the methods scaladoc maybe